apps/prairielearn/src/components/CommentPopover.html.ts

```diff
@@ -0,0 +1,32 @@
+import { escapeHtml, html } from '@prairielearn/html';
+
+export function CommentPopover(comment: string | string[] | Record<string, any> | null) {
+  let content = '';
+  if (
+    !comment ||
+    (typeof comment === 'string' && comment.trim() === '') ||
+    (Array.isArray(comment) && comment.toString() === '[]') ||
+    (typeof comment === 'object' && Object.keys(comment).length === 0)
+  ) {
+    return '';
+  } else if (typeof comment === 'string') {
+    content = comment;
+  } else {
+    content = JSON.stringify(comment, null, 2);
+  }
```

Comment:
Nit, this could be written more compactly:

```suggestion
  }
  
  const content = typeof comment === 'string' ? comment : JSON.stringify(comment, null, 2);
```